### PR TITLE
107 fix warnings in browser console when show 404 page

### PR DESCRIPTION
**Pull request by VitalyyP**
## Release Notes

- **New Features**
	- Introduced a new `Custom404` component for a more streamlined 404 error page experience.
	- Enhanced error handling for product pages, redirecting users to a not-found page when product data is missing.
	- Added modal functionality to display fabric options when multiple colors are available in the product details wizard.
	- Improved accessibility and user interaction in various components with updated input elements and labels.

- **Bug Fixes**
	- Improved metadata generation for product pages to ensure default values are applied when necessary.
	- Added null checks across various components to prevent errors when product data is unavailable.

- **Style**
	- Updated styles for the `Custom404` page to enhance visual presentation, including adjustments to font sizes, margins, and layout.
	- Adjusted image height for better responsiveness on the 404 error page.
	- Enhanced styling for input elements and placeholders to ensure consistency across the application.

- **Chores**
	- Removed the old 404 page component to simplify the codebase.
	- Commented out unnecessary redirection in the ShopPage function to improve clarity and maintainability.
	- Disabled the end-to-end testing phase in the workflow for potential reactivation in the future.

### CODE REVIEW

File: src/app/[lang]/shop/[categoryId]/[productId]/layout.tsx [hunk 1]
```diff
@@ -26,14 +27,25 @@ export async function generateMetadata({ params }: Props): Promise<Metadata> {
...
+  let title = 'Product Not Found';
+  let description = 'This product does not exist.';
+  try {
+    const data = (await getDictionary(lang, 'meta')) as metaI18N;
+    productMeta = data?.[productId];
+    title = productMeta.title;
+    description = productMeta.description;
+  } catch (error) {
+    notFound();
+  }
```
[the remark this answers is not in the document]
Comment (VitalyyP) [reply src/app/[lang]/shop/[categoryId]/[productId]/layout.tsx]: I think for this case this is better. This error is a part of logic.

File: src/data/viewModels/product/productDetailsPrevSliderVM.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ function generateImagePath(dirname, id, num = 1) {
 }
 
 const generateGalleryForProduct = (product) => {
+  if (!product) return null;
+
```
[the remark this answers is not in the document]
Comment (VitalyyP) [reply src/data/viewModels/product/productDetailsPrevSliderVM.js:10]: I use if(!product) to render nothing.

File: src/components/productPage/infoSection/InfoSection.tsx [hunk 3]
```diff
@@ -17,7 +17,9 @@ type Props = {
 
 function InfoSection({ productId, language }: Props) {
   const product = getProductById(productId);
-  const { embedVideo } = product;
+  if (!product) return null;
+
+  const embedVideo = product.embedVideo;
```
[the remark this answers is not in the document]
Comment (VitalyyP) [reply src/components/productPage/infoSection/InfoSection.tsx:22]: I use if(!product) to render nothing.

**Review by VolodymyrVorona — DISMISSED** [review]
LGTM

**Pull request merged**